lib/cli/commands/listswaps.ts

```diff
@@ -0,0 +1,93 @@
+import { callback, loadXudClient } from '../command';
+import { Arguments } from 'yargs';
+import { ListSwapsRequest, ListSwapsResponse } from '../../proto/xudrpc_pb';
+import Table , { HorizontalTable } from 'cli-table3';
+import colors from 'colors/safe';
+
+const FAILED_HEADERS = [
+  colors.blue('peer public key'),
+  colors.blue('order id'),
+  colors.blue('trading pair'),
+  colors.blue('quantity'),
+  colors.blue('failure reason'),
+];
+// TODO: create a function that generates these tables
+const SUCCESS_HEADERS = [
+  colors.blue('peer public key'),
+  colors.blue('order id'),
+  colors.blue('local id'),
+  colors.blue('trading pair'),
+  colors.blue('quantity'),
+  colors.blue('price'),
+  colors.blue('amount sent'),
+  colors.blue('amount received'),
+  colors.blue('currency received'),
+  colors.blue('currency send'),
+  colors.blue('role'),
+  colors.blue('rhash'),
+  colors.blue('rpreimage'),
+];
+
+const displaySwaps = (swaps: ListSwapsResponse.AsObject) => {
+  const failedTable = new Table({ head: FAILED_HEADERS }) as HorizontalTable;
+  const successTable = new Table({ head: SUCCESS_HEADERS }) as HorizontalTable;
+
+  swaps.swapsList.forEach((swap) => {
+    if (swap.swapFailure) {
+      const fail = swap.swapFailure;
+      failedTable.push([
+        fail.peerPubKey,
+        fail.orderId,
+        fail.pairId,
+        fail.quantity,
+        fail.failureReason,
+      ]);
+    } else {
+      const success = swap.swapSuccess!;
+      successTable.push([
```

Comment:
I suspect these are going to be too many columns for a horizontal table in a terminal. I'm thinking we should probably drop:

- preimage
- pairId (assuming we keep currency received and sent)
- orderId (assuming we keep localId)

I'd also consider dropping quantity & price (assuming we keep amounts sent and received).

Thoughts @kilrau ?

Reply:
Sounds good.